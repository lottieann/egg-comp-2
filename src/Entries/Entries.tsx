import * as React from 'react'

import styled from 'styled-components'

import Image from 'next/image'
import Link from 'next/link'

import { Entry } from '../types'

export interface EntriesProps {
  entries: {
    entryNumber: number;
    title: string;
    thumbnail: string;
    photo1?: string;
    photo2?: string;
    photo3?: string;
    videoLink?: string;
    additional?:string;
  }
}

const Container = styled.div`
  margin-top: 15vh;
  display: flex;
  flex-direction: column;
  align-items: center;

  h1 {
    font-family: Helvetica;
  }
`

const EntryList = styled.div`
  display: flex;
  flex-wrap: wrap;
  padding: 20px;
`

const EntryContainer = styled.div`
  margin: 2vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  cursor: pointer;

  p {
    font-family: Helvetica;
    font-weight: bold;
  }
`


const StyledAnchor = styled.a`
  text-decoration: none;
  font-weight: bold;
  font-family: Helvetica;
  cursor: pointer;
  padding: 2vh;
`

export const Entries = (props: EntriesProps) => {
  
  const makeEntriesList = (options: Entry) => {
    return (
      <EntryList>
        {Object.values(options).map((el, index) => {
          return (
             //@ts-ignore
            el.entryNumber !== 0 &&
            <EntryContainer key={index}>
                <Link href={`/entries/${index}`} key={index}>
                  <Image
                    data-testid={`image-${index}`}
                    //@ts-ignore
                    src={`/images/${el.thumbnail}.jpg`}
                    alt={`egg-${index}`}
                    width={400}
                    height={265}
                  />
                </Link>
              <Link href={`/entries/${index}`}>
              {/* @ts-ignore */}
                <StyledAnchor key={index}>{el.entryNumber}</StyledAnchor>
              </Link>
              {/* @ts-ignore */}
              <p>{el.title}</p>
            </EntryContainer>
          )
        })}
      </EntryList>
    )
  }

  return (
    <Container>
      <h1>Select an entry to view</h1>
      <ul>{makeEntriesList(props.entries)}</ul>
    </Container>
  )
}
