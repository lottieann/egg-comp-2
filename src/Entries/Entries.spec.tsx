import * as React from 'react'

import 'jest-styled-components'
import '@testing-library/jest-dom/extend-expect'

import { render } from '@testing-library/react'
import { Entries, EntriesProps } from './Entries'

const entries = [
  {
    entryName: 'Test Entry',
    entryNumber: 0,
    photo1: 'egg-comp-app/public/0.jpg',
    photo2: 'egg-comp-app/public/1.jpg',
    photo3: 'egg-comp-app/public/3.jpg',
    vote1: '',
    vote2: '',
    vote4: '',
    vote3: '',
    vote5: '',
  },
  {
    entryName: 'Test Entry 1',
    entryNumber: 1,
    photo1: 'egg-comp-app/public/0.jpg',
    photo2: 'egg-comp-app/public/1.jpg',
    photo3: 'egg-comp-app/public/3.jpg',
    vote1: '',
    vote2: '',
    vote4: '',
    vote3: '',
    vote5: '',
  },
  {
    entryName: 'Test Entry 2',
    entryNumber: 2,
    photo1: 'egg-comp-app/public/0.jpg',
    photo2: 'egg-comp-app/public/1.jpg',
    photo3: 'egg-comp-app/public/3.jpg',
    vote1: '',
    vote2: '',
    vote4: '',
    vote3: '',
    vote5: '',
  },
]


const setup = (props: EntriesProps) => {
  return <Entries {...props}/>
}

describe('Entries', () => {
  it('should render entries correctly', () => {
    const { container } = render(setup({
      entries
    }))

    expect(container).toMatchSnapshot()
  })
})