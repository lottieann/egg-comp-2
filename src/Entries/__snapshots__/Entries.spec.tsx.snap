// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Entries should render entries correctly 1`] = `
.c0 {
  margin-top: 15vh;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

<div>
  <div
    class="c0"
  >
    <p>
      Entries will be able to view on Easter Satruday
    </p>
  </div>
</div>
`;
